src/suas_planning/src/GlobalWaypointPlanner.cpp

```diff
@@ -98,28 +211,43 @@ int GlobalWaypointPlanner::UpdateMap(std::vector<Obstacle>& obstacles) {
     return obstacles.size();
 }
 
-GlobalWaypointPlanner::Node::Node(int x, int y, Waypoint& goal, int action, Node& from, GlobalWaypointPlanner& planner) :
-    x_(x),
-    y_(y),
-    action_(action),
-    goal_(goal),
-    from_(from),
-    g_cost_(action_cost[action]),
-    parent_(planner) {
+GlobalWaypointPlanner::Node::Node(
+    int x,
+    int y,
+    const Waypoint& goal,
+    int action,
+    std::shared_ptr<Node> from,
+    GlobalWaypointPlanner& planner) :
+        x_(x),
+        y_(y),
+        action_(action),
+        goal_(goal),
+        from_(from),
+        parent_(planner) { 
+    if (action != NO_ACTION) {
+        g_cost_ = action_cost[action];
         h_cost_ = ComputeHeuristicCost();
         f_cost_ = g_cost_ + h_cost_;
+    } else {
+        // Special case for start node
+        f_cost_ = h_cost_ = g_cost_ = 0;
+    }
 }
 
-double GlobalWaypointPlanner::Node::ComputeHeuristicCost() {
-    double dx = goal_.x_ - x_;
-    double dy = goal_.y_ - y_;
+double GlobalWaypointPlanner::Node::ComputeHeuristicCost() { 
+    double dx = (double) (goal_.x_ - x_);
```

Comment:
In C++, one should lean towards using the typed *_cast<> family for casting. The short answer is that it can ignore type safety and let you shoot yourself in the foot. Longer answer can be found at [StackOverflow](https://stackoverflow.com/a/103868).

Reply:
Probably could regex search for these because they're throughout the codebase. static_cast<type> is pretty much the go-to unless there's a special case

Reply:
Resolved in 5ba2064